### DISCUSSION TITLE
how to read a record out of graphcache?

### DISCUSSION THREAD

**Discussion by videni** [urql-graphql/urql#3260] [Q&A]
I enabled the graphcache exchange, a page queried a  record already and cached,  how can I get that in another page? I am not able to pass the whole record data to another page, just only the record id,  Do I  have to  execute the query again?   any alternatives please?

**Comment by kitten** [comment]
You're looking for this part of the docs: https://formidable.com/open-source/urql/docs/graphcache/normalized-caching/#manually-resolving-entities

**Comment by videni** [comment]
@kitten , thanks,  for a total newbie of uqrl,  the doc is so overwhelming, I read  them all, but I still don't get it. here are what I did, 
 
# 1. Add `ceramicaCollection` resolver

```
const client = createClient({
  url: 'http://localhost:8000/graphql',
  exchanges: [
    fetchExchange,
    cacheExchange({
      resolvers: {
        Query: {
          ceramicaCollection(parent, args, cache, info) {
            return { __typename: 'Entry_CeramicaCollections_CeramicaCollection', id: args.id };
          },
        },
      },
    })
  ],
});
```
# 2. Query field `ceramicaCollection`
```
  let [result1, reexecuteQuery1] = useQuery({
    query: gql`
        query ($id: String){
          ceramicaCollection(id: $id){
            __typename
            id
          }
      }
    `,
    variables: {
      id: colId
    }
  })

  console.log(result1);
```
# 3. Got error:

```
"[GraphQL] Cannot query field \"ceramicaCollection\" on type \"Query\". Did you mean \"collection\"?".  
```

this comlains might be that my server schema doesn't have field `ceramicaCollection`, but since I created a local one, I don't know why.

# 4. The record queried by following 

```
export const CollectionQuery = gql`
  query ($id: String){
    entry(id: $id, collection: "ceramica_collections"){
       id
      title
      content
      logo {
         permantlink
     }
    }
  }
`

 let [result, reexecuteQuery] = useQuery({
    query: CollectionQuery,
    variables: {
      id: colId
    }
  })
```
# Questions 

## 1. I just need to get the cached data out of graphcache, Why we need to add a similar local resolver `ceramicaCollection`  which is the same as the server side `collection` field, any easier way?

**Reply by kitten** [reply]
The cache in urql will always reflect your remote schema, so if your API doesn't have a field you can't query it.

In general, in GraphQL you want your local data to basically always be a representation of your remote data, so you have to update your schema if you want to access your API in a new way
